src/main/java/de/teamlapen/vampirism/client/gui/MinionStatsScreen.java

```diff
@@ -0,0 +1,106 @@
+package de.teamlapen.vampirism.client.gui;
+
+import de.teamlapen.lib.lib.util.UtilLib;
+import de.teamlapen.vampirism.VampirismMod;
+import de.teamlapen.vampirism.entity.minion.MinionEntity;
+import de.teamlapen.vampirism.entity.minion.management.MinionData;
+import de.teamlapen.vampirism.network.UpgradeMinionStatPacket;
+import de.teamlapen.vampirism.util.REFERENCE;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public abstract class MinionStatsScreen<T extends MinionData, Q extends MinionEntity<T>> extends Screen {
+
+    private static final ResourceLocation BACKGROUND = new ResourceLocation(REFERENCE.MODID, "textures/gui/appearance.png");
+
+
+    protected final Q entity;
+    protected final int xSize = 256;
+    protected final int ySize = 177;
+    protected final int statCount;
+    private final TranslationTextComponent textLevel = new TranslationTextComponent("text.vampirism.level");
+    private final List<Button> statButtons = new ArrayList<>();
+    protected int guiLeft;
+    protected int guiTop;
+
+    protected MinionStatsScreen(Q entity, int statCount) {
+        super(new TranslationTextComponent("gui.vampirism.minion_stats"));
+        assert statCount > 0;
+        this.entity = entity;
+        this.statCount = statCount;
+    }
+
+
+    @Override
+    public void render(int mouseX, int mouseY, float partialTicks) {
+        this.renderBackground();
+        this.renderGuiBackground();
+        this.drawTitle();
+        super.render(mouseX, mouseY, partialTicks);
+        entity.getMinionData().ifPresent(this::renderStats);
+
+    }
+
+    @Override
+    public void tick() {
+        for (int i = 0; i < statCount; i++) {
+            int finalI = i;
+            statButtons.get(i).active = entity.getMinionData().map(d -> isActive(d, finalI)).orElse(false);
+            statButtons.get(i).visible = entity.getMinionData().map(this::areButtonsVisible).orElse(false);
+        }
+
+    }
+
+    @Override
+    protected void init() {
+        this.guiLeft = (this.width - this.xSize) / 2;
+        this.guiTop = (this.height - this.ySize) / 2;
+
+        this.addButton(new Button(this.guiLeft + ((this.xSize - 80) / 2), this.guiTop + 152, 80, 20, UtilLib.translate("gui.done"), (context) -> {
+            this.onClose();
+        }));
+        for (int i = 0; i < statCount; i++) {
+            int finalI = i;
+            statButtons.add(this.addButton(new Button(guiLeft + 225, guiTop + 43 + 26 * i, 20, 20, "+", (button) -> VampirismMod.dispatcher.sendToServer(new UpgradeMinionStatPacket(entity.getEntityId(), finalI)))));
```

Comment:
make them invisible on creation.

if the minion is full level the buttons are shown for a brief moment before hiding them